app/js/S3PersistorManager.js

```diff
@@ -240,9 +249,49 @@ function _wrapError(error, message, params, ErrorType) {
   }
 }
 
-function _defaultOptions() {
-  const options = {
-    credentials: {
+const _clients = {}
```

Comment:
Could this use a `Map`? I think it is best to move away from using objects as maps where possible, now that we have maps.

Reply:
Yes, this sounds like a good habit to cultivate.